src/api.ts

```diff
@@ -25,3 +26,34 @@ function get<T>(url: string): Promise<T> {
 export const getTodos = () => get<Todo[]>('/todos');
 
 export const getUser = (userId: number) => get<User>(`/users/${userId}`);
+
+export const filterTodo = (
+  todos: Todo[],
+  filterParam: Filter,
+  queryParam: string,
+) => {
+  let todosCopy = [...todos];
+
+  switch (filterParam) {
+    case Filter.Active:
+      todosCopy = todosCopy.filter(todo => !todo.completed);
+      break;
+    case Filter.Completed:
+      todosCopy = todosCopy.filter(todo => todo.completed);
+      break;
+    case Filter.All:
+    default:
+      break;
+  }
+
+  if (queryParam) {
+    todosCopy = todosCopy.filter(todo => {
+      const lowerQueryParam = queryParam.toLowerCase();
+      const lowerTitle = todo.title.toLowerCase();
+
+      return lowerTitle.includes(lowerQueryParam);
+    });
+  }
```

Comment:
if query - return filtered values
```suggestion
  if (queryParam) {
    return visibleTodos .filter(todo => {
      const lowerQueryParam = queryParam.toLowerCase();
      const lowerTitle = todo.title.toLowerCase();

      return lowerTitle.includes(lowerQueryParam);
    });
  }
```